test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -2193,6 +2201,178 @@ Describe "Invoke-WebRequest tests" -Tags "Feature", "RequireAdminOnWindows" {
             $pathologicalRatio | Should -BeGreaterThan 5
         }
     }
+
+    Context 'WebSession persistence tests' {
+        $connWarning = 'The WebSession properties were changed between requests forcing all HTTP connections in the session to be recreated.'
+
+        It 'Connection persistence maintained' {
+            $uri = Get-WebListenerUrl
+            $verboseMessages = Invoke-WebRequest -Verbose -Uri $uri -SessionVariable Session 4>&1 | FilterVerboseMessages -Message $connWarning
+            $verboseMessages | Should -BeNullOrEmpty
+            # Multiple follow-up requests using the same web session are persistent
+            1 .. 3 | ForEach-Object {
+                $verboseMessages = Invoke-WebRequest -Verbose -Uri $uri -WebSession $Session 4>&1 | FilterVerboseMessages -Message $connWarning
+                $verboseMessages | Should -BeNullOrEmpty
+            }
+        }
+
+        It 'Connection persistence outperforms non-persistence' {
+            $uri = Get-WebListenerUrl -Https
+            $null = Invoke-WebRequest -Uri $uri -SessionVariable Session -SkipCertificateCheck
+            $iters = 5
+            $persistence = Measure-Command {
+                1 .. $iters | ForEach-Object {
```

Comment:
I am ok to remove the test. Only please add perf comparisons in the PR description.

Reply:
Ok, the tests are very fast when running with the mock service, since it's all localhost. The connection setup, even for SSL is pretty quick also, since we have to ignore certificate errors. A comparison with a real HTTPS site with certificate checking gives much bigger performance improvement from persistence.

I will update the PR with some benchmarks against the postman site as per the issue.